services/src/main/scala/cromwell/services/instrumentation/impl/stackdriver/StackdriverConfig.scala

```diff
@@ -0,0 +1,69 @@
+package cromwell.services.instrumentation.impl.stackdriver
+
+import cats.data.Validated._
+import cats.syntax.apply._
+import cats.syntax.validated._
+import com.typesafe.config.Config
+import common.exception.MessageAggregation
+import common.validation.ErrorOr.ErrorOr
+import common.validation.Validation.validate
+import cromwell.cloudsupport.gcp.GoogleConfiguration
+import cromwell.cloudsupport.gcp.auth.{GoogleAuthMode, UserServiceAccountMode}
+import net.ceedubs.ficus.Ficus._
+
+import scala.concurrent.duration._
+
+case class StackdriverConfig(googleProject: String,
+                             auth: GoogleAuthMode,
+                             flushRate: FiniteDuration,
+                             cromwellInstanceIdentifier: Option[String],
+                             cromwellInstanceRole: Option[String],
+                             cromwellPerfTestCase: Option[String])
+
+object StackdriverConfig {
+  val CromwellInstanceIdentifier = "cromwell-instance-identifier"
+  val CromwellInstanceRole = "cromwell-instance-role"
+  val CromwellPerfTest = "cromwell-perf-test-case"
+
+  private def validateFlushRate(rate: FiniteDuration): ErrorOr[FiniteDuration] = {
+    validate[FiniteDuration](rate) match {
+      case Valid(duration) => duration match {
+        case d: FiniteDuration if d.<(1.minute) => (s"`flush-rate` can't be less than 1 minute. Current rate is `${d.toString}`. " +
```

Comment:
@mcovarr if the config is invalid, the error returned is `Stackdriver instrumentation config is invalid. Errors(s): `flush-rate`...`. It would be redundant to add `*Stackdriver instrumentation* `flush-rate`` again in the error message. I have updated the remaining part of the error message from your suggestion.

@kshakir I have updated the logic with your suggestions.

Thank you both!